src/clink.rs

```diff
@@ -35,7 +35,7 @@ impl Clink {
         let mut res = str.to_string();
         for link in self.finder.links(str) {
             let mut l = Url::parse(self.unwrap_exit_params(link.as_str()).as_str()).unwrap();
-            let query: Vec<(_, _)> = self.process_query(l.query_pairs());
+            let query: Vec<(_, _)> = self.process_query(l.query_pairs(),l.domain());
```

Comment:
`l.domain()` will return the domain name, including `www`. 

This means URLs like `https://www.youtube.com/watch?v=dQw4w9WgXcQ&si=NblIBgit-qHN7MoH` will not be sanitized with the current config. 

It is a good idea to use a library like https://crates.io/crates/addr or https://crates.io/crates/publicsuffix to extract registrable domains.